hs_tracking/views.py

```diff
@@ -24,6 +24,15 @@ def get(self, request, **kwargs):
 
         if not authentic_redirect_url(url):
             return HttpResponseForbidden()
+
+        # encode url placeholder values received from the front-end
+        url_placeholders = ["HS_JS_AGG_KEY", "HS_JS_MAIN_FILE_KEY", "HS_JS_FILE_KEY"]
+        for placeholder in url_placeholders:
+            placeholder_value = querydict.pop(placeholder, [''])[0]
+            if placeholder in url:
+                encoded_value = urllib.parse.quote(placeholder_value)
```

Comment:
my only issue here is edge cases like filenames that already contain "%20" for example but that seems improbable...

Not sure if using [quote_plus](https://docs.python.org/3/library/urllib.parse.html#urllib.parse.quote_plus) or similar would help us catch some of these edge cases, but probably not worth our time to stress it too much?

Reply:
Potentially we will have encoding problems as currently we don't have any restriction on filename and folder name. I don't think using `quote_plus` will help us avoid such problems. I think we need to first put some restrictions on filename. We have talked about filename restrictions before but have not reached at any solution yet.

Reply:
sounds good